### PR TITLE
JKNS-257: add e2es for java/nodejs sample sidecar pod templates

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan 
/assign @akram 

these new test should fail until https://github.com/openshift/jenkins/pull/1355 merges

### CODE REVIEW

File: test/e2e/sync_plugin_test.go [hunk 1]
```diff
@@ -257,6 +257,46 @@ func podTemplateTest(name string, ta *testArgs) {
 
 }
 
+func javaBuilderPodTemplateTest(name string, ta *testArgs) {
+	bc := &buildv1.BuildConfig{}
+	bc.Name = name
+	bc.Spec = buildv1.BuildConfigSpec{
```
Comment (adambkaplan) [inline test/e2e/sync_plugin_test.go:263]: Are we doing any `gofmt` linting on this project? Two spaces instead of tabs seems way out of standard go format.

File: test/e2e/testobjects.go [hunk 2]
```diff
@@ -52,6 +52,43 @@ const (
...
+                        
+                    }
+                }
+             }
+          } catch (err) {
+             echo "in catch block"
+             echo "Caught: ${err}"
+             currentBuild.result = 'FAILURE'
+             throw err
+          }
```
Comment (adambkaplan) [inline test/e2e/testobjects.go:72]: Any particular reason why this is indented this far? Typically I don't see indents for golang string literals

File: test/e2e/testobjects.go [hunk 3]
```diff
@@ -52,6 +52,43 @@ const (
...
+`
```
Comment (adambkaplan) [inline test/e2e/testobjects.go:91]: Same wrt indents

**Review by adambkaplan — COMMENTED** [review]
Mostly cosmetic concerns. My initial issue with `gofmt ` should be addressed in a separate PR.

**Comment by adambkaplan** [comment]
/retest

Got bit by AWS outage earlier today.

**Comment by gabemontero** [comment]
> /retest
> 
> Got bit by AWS outage earlier today.

Per the description the new test here won't pass until the Jenkins PR merges  

This is one of the edge cases of including the e2e's in the plugin vs. the image repo.

I've verified the new tests manually, but if this catches something after the jenkins repo PR merge, we'll address accordingly

File: test/e2e/sync_plugin_test.go (hunk 1 above)
Comment (gabemontero) [reply test/e2e/sync_plugin_test.go:263]: yes this has come up before ... I'm running `gofmt` .... not sure what is going on here

I'll post a specific before / after example later on.  And if as goland upgrade as addressed it, then yes, we'll do it in a separate PR :-)

File: test/e2e/testobjects.go (hunk 2 above)
Comment (gabemontero) [reply test/e2e/testobjects.go:72]: the jenkins console I ran this test from to confirm function indented it and copied form there directly

I can try to manually remove indents if need be

**Comment by gabemontero** [comment]
On the go formatting @adambkaplan .... yeah, here is what happens for me:

```
gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ git status 
On branch add-sample-side-car-e2es
nothing to commit, working tree clean
gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ gofmt -w ./test/
gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ git status 
On branch add-sample-side-car-e2es
nothing to commit, working tree clean
gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ 
```

maybe there is some setting in that repo for me I am unaware of the is messing with the indenting?

I'd be curious what you or @akram get when you try this in your clones of master branch

**Comment by gabemontero** [comment]
ok @akram @adambkaplan the new tests for the new pod templates are passing, I shifted the new test pipeline groovy stanzas to the left, and we've agreed for someone besides me to run `gofmt` and the test directory after this merges

more review, or are y'all ready to tag for merge ?

**Comment by gabemontero** [comment]
by the way @akram @adambkaplan I believe @alicerum will be adding e2e's as part of her upcoming fix, so ideally we get this merged before she starts in earnest so she can make changes on top of this.

**Comment by alicerum** [comment]
@gabemontero i have already added the e2e test, but i'll rebase as soon as this gets merged

**Comment by gabemontero** [comment]
On Fri, Dec 10, 2021 at 9:58 AM Alice Rum ***@***.***> wrote:

> @gabemontero <https://github.com/gabemontero> i have already added the
> e2e test, but i'll rebase as soon as this gets merged
>
ah in that case @alicerum if you get the fix PR up soon we can merge that
and I can rebase this after as well

Your BZ takes precedence

File: test/e2e/sync_plugin_test.go (hunk 1 above)
Comment (gabemontero) [reply test/e2e/sync_plugin_test.go:263]: with @adambkaplan 's awesome fine about .editorconfig the tabbing now looks normal in this PR ...check it out

**Comment by gabemontero** [comment]
Still no rebase warning after @alicerum 's PR merged

this PR is ready @adambkaplan @akram for any final comments / tag for merge
